test/unit/puppetlabs/services/jruby/jruby_puppet_service_test.clj

```diff
@@ -289,3 +289,128 @@
+         (is (= "1234prod" (jruby-protocol/get-environment-class-info-tag
+                            service
+                            "production")))
+         (is (nil? (jruby-protocol/get-environment-class-info-tag
+                    service
+                    "test")))
+         (jruby-protocol/set-environment-class-info-tag! service
+                                                         "test"
+                                                         "1234test"
+                                                         nil)
+         (is (= "1234prod" (jruby-protocol/get-environment-class-info-tag
+                            service
+                            "production")))
+         (is (= "1234test" (jruby-protocol/get-environment-class-info-tag
+                            service
+                            "test"))))
+       (let [production-first-update
+             (jruby-protocol/get-environment-class-info-tag-last-updated
+              service
+              "production")
+             test-first-update
+             (jruby-protocol/get-environment-class-info-tag-last-updated
+              service
+              "test")]
+         (testing "when environment info reset in the cache"
+           (is (not (nil? production-first-update)))
+           (is (not (nil? test-first-update)))
+           (jruby-protocol/set-environment-class-info-tag!
+            service
+            "production"
+            "5678prod"
+            production-first-update)
+           (is (= "5678prod" (jruby-protocol/get-environment-class-info-tag
+                              service
+                              "production")))
+           (is (= "1234test" (jruby-protocol/get-environment-class-info-tag
+                              service
+                              "test")))
+           (is (= test-first-update
+                  (jruby-protocol/get-environment-class-info-tag-last-updated
+                   service
+                   "test")))
+           (testing "and environment expired between get and corresponding set"
+             (let [production-second-update
+                   (jruby-protocol/get-environment-class-info-tag-last-updated
+                    service
+                    "production")
+                   _ (jruby-protocol/mark-environment-expired! service
+                                                               "production")
+                   production-third-update
+                   (jruby-protocol/get-environment-class-info-tag-last-updated
+                    service
+                    "production")]
+               (is (not= production-first-update production-second-update))
+               (jruby-protocol/set-environment-class-info-tag!
+                service
+                "production"
+                "89abprod"
+                production-second-update)
+               (is (= nil (jruby-protocol/get-environment-class-info-tag
```

Comment:
Having just reviewed this PR, I understand why this is `nil` but the last-updated timestamp is `production-third-update`. However, I think that if I were to come back to these tests in a few weeks without that context I might be very confused by these assertions. It might help to add some other `(testing` blocks or comments to add description.